spec/helpers/application_helper/buttons/auth_key_pair_cloud_create_spec.rb

```diff
@@ -2,9 +2,8 @@
   let(:button) { described_class.new(setup_view_context_with_sandbox({}), {}, {}, {}) }
 
   def setup_ems(supports)
-    ems = object_double(ManageIQ::Providers::CloudManager.new)
-    allow(ems).to receive(:supports?).with(:auth_key_pair_create).and_return(supports)
-    allow(Rbac).to receive(:filtered).and_return([ems])
+    ems = FactoryBot.create(:ems_cloud)
+    allow(ManageIQ::Providers::CloudManager).to receive(:where).and_return(supports ? [ems] : [])
```

Comment:
is it possible to stub_supports here instead of hacking the where?

sigh. we really need a cloud and infra dummy providers.

Reply:
> we really need a cloud and infra dummy providers.

Been discussing this independently with @agrare and I'm considering bringing in the dummy provider as an actual provider: https://github.com/ManageIQ/manageiq-providers-dummy_provider

Reply:
> is it possible to stub_supports here instead of hacking the where?

:+1: yeah good call stubbing the where is pretty hacky

Reply:
Thanks for the nudge @kbrock, this came out looking way better